### PR TITLE
feat: Add "Total Remaining Budget" display

### LINKED ISSUE #3365: [TOTAL REMAINING BUDGET]
### Please confirm the following:

- [X] I've checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.
- [X] I've requested a single (only one) feature/change in this issue. It complies with the One Request Per GitHub Issue (ORPGI) rule.
- [X] My issue is well-defined and describes how it should be implemented from UI/UX perspective.

### What do you want to be added or improved?

As a user I want to know how much my budget left, not only the summary of my planned budget.

### Why do you need it?

A. From there I can find out how much my budget left straightly, no need to summarize it manually anymore
B. And so from there I can find out whether the remaining budget exceeds the total remaining money or not

### How do you imagine it?

The total remaining budget will be shown beneath the budget info, described as "Total Remaining Budget"

### PR DESCRIPTION
**Pull request by akashs056**
## Pull request (PR) checklist
Please check if your pull request fulfills the following requirements:

- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [x] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## Screen recording of interacting with your changes:

https://github.com/user-attachments/assets/b2ad6d2b-a159-4f51-ba60-80ce8fa4c500

## What's changed?
 - Added display for "Total Remaining Budget" beneath budget information. 
 - Calculated remaining budget for categorized and multi-categorized budgets.
 - Included multi-language support for the "Total Remaining Budget" text. 
 - Enhanced user experience by providing a clear overview of remaining budget without manual summarization.

## Does this PR close any GitHub issues? (do not delete)
- Closes #3365

### CODE REVIEW

File: screen/budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt [hunk 1]
```diff
@@ -176,6 +183,10 @@ class BudgetViewModel @Inject constructor(
                     )
                 }.toImmutableList()
             }
+            this@BudgetViewModel.totalRemainingBudget.doubleValue = calculateTotalRemainingBudget(
```
Comment (ILIYANGERMANOV) [inline screen/budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt:186]: no need for `this@`, if there's a clash of names rename to `totalRemainingBudgetState`

File: screen/budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt [hunk 2]
```diff
@@ -262,3 +273,12 @@ class BudgetViewModel @Inject constructor(
         }
     }
 }
+
+    fun calculateTotalRemainingBudget(
+        budgets: ImmutableList<DisplayBudget>,
+        categoryBudgetsTotal: Double
+    ): Double {
+        return categoryBudgetsTotal - budgets
+            .filter { !it.budget.categoryIdsSerialized.isNullOrBlank() }
```
Comment (ILIYANGERMANOV) [inline screen/budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt:282]: use `isNotNullOrBlank()` and also add a comment that you're filtering to only Category budgets

File: screen/budgets/src/main/java/com/ivy/budgets/BudgetScreen.kt [hunk 3]
```diff
@@ -169,11 +171,13 @@ private fun BoxWithConstraintsScope.UI(
 @Composable
 private fun Toolbar(
     timeRange: com.ivy.legacy.data.model.FromToTimeRange?,
+    totalRemainingBudget: Double,
     baseCurrency: String,
     appBudgetMax: Double,
     categoryBudgetsTotal: Double,
     setReorderModalVisible: (Boolean) -> Unit
 ) {
+    val budgetExceeded = totalRemainingBudget < 0
```
Comment (ILIYANGERMANOV) [inline screen/budgets/src/main/java/com/ivy/budgets/BudgetScreen.kt:180]: Move `budgetExceeded` as close as possible to its usages

File: screen/budgets/src/main/java/com/ivy/budgets/BudgetScreen.kt [hunk 4]
```diff
@@ -244,6 +248,25 @@ private fun Toolbar(
                         fontWeight = FontWeight.ExtraBold
                     )
                 )
+
+                val totalRemainingText = when {
```
Comment (ILIYANGERMANOV) [inline screen/budgets/src/main/java/com/ivy/budgets/BudgetScreen.kt:252]: This transformation should be done in the VM. Change the UI state to be `: String?` and do this calculation and formating in the VM

File: screen/budgets/src/main/java/com/ivy/budgets/BudgetScreen.kt [hunk 5]
```diff
@@ -244,6 +248,25 @@ private fun Toolbar(
...
+                    categoryBudgetText.isNotEmpty() -> stringResource(
+                        if (budgetExceeded) R.string.budget_exceeded_info else R.string.total_budget_info,
+                        abs(totalRemainingBudget).format(baseCurrency),
+                        baseCurrency
+                    )
+                    else -> ""
+                }
+
+                if (categoryBudgetText.isNotEmpty()) {
```
Comment (ILIYANGERMANOV) [inline screen/budgets/src/main/java/com/ivy/budgets/BudgetScreen.kt:261]: Is this `if` correct? I guess we need to check `totalRemaingText` not being null

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
Overall looks good, a few changes:
- use `String?` for the UI state of the new text 
- do the calculation and formatting in the VM

This is needed for performance and architecture reasons.

**Comment by ILIYANGERMANOV** [comment]
Ping me for re-review when you've applied the changes

**Comment by akashs056** [comment]
@ILIYANGERMANOV applied the changes. please re-review it

File: screen/budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt [hunk 6]
```diff
@@ -62,6 +66,8 @@ class BudgetViewModel @Inject constructor(
     private val accounts = mutableStateOf<ImmutableList<Account>>(persistentListOf())
     private val categoryBudgetsTotal = mutableDoubleStateOf(0.0)
     private val appBudgetMax = mutableDoubleStateOf(0.0)
+    private val totalRemainingBudget = mutableDoubleStateOf(0.0)
+    private val totalRemainingBudgetText = mutableStateOf<String?>("")
```
Comment (ILIYANGERMANOV) [inline screen/budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt:70]: We don't need this state - just transform the double state inside `getTotalRemainingBudgetText()`

File: screen/budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt [hunk 7]
```diff
@@ -124,6 +131,20 @@ class BudgetViewModel @Inject constructor(
         return appBudgetMax.doubleValue
     }
 
+    @Composable
+    private fun getTotalRemainingBudgetText(): String? {
+        val budgetExceeded = totalRemainingBudget.doubleValue < 0
+        totalRemainingBudgetText.value = when {
```
Comment (ILIYANGERMANOV) [inline screen/budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt:137]: Remove the unnecessary text state and just put `return` here

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
Remove unnecessary state, fix Detekt and we can merge

**Comment by akashs056** [comment]
its ready to merge

**Review by ILIYANGERMANOV — APPROVED** [review]

**Pull request merged**